src/esp32/pwm.rs

```diff
@@ -6,31 +7,21 @@ use esp_idf_hal::ledc::{
 };
 use esp_idf_hal::peripheral::Peripheral;
 use esp_idf_hal::prelude::FromValueType;
-use esp_idf_sys::{
-    ledc_bind_channel_timer, ledc_channel_t_LEDC_CHANNEL_MAX, ledc_get_freq, ledc_set_freq,
-    ledc_timer_t, EspError,
-};
+use esp_idf_sys::{ledc_bind_channel_timer, ledc_get_freq, ledc_timer_t, EspError};
 use once_cell::sync::Lazy;
-use std::collections::HashMap;
+use std::cell::OnceCell;
+use std::fmt::Debug;
 use std::sync::Mutex;
 use thiserror::Error;
 
 static LEDC_MANAGER: Lazy<Mutex<LedcManager>> = Lazy::new(|| Mutex::new(LedcManager::new()));
 
-pub(crate) fn create_pwm_driver(
-    pin: AnyIOPin,
-    starting_frequency_hz: u32,
-) -> Result<PwmDriver<'static>, Esp32PwmError> {
-    let chan = LEDC_MANAGER.lock().unwrap().take_next_channel(pin.pin())?;
-    PwmDriver::new(pin, chan, starting_frequency_hz)
-}
-
 #[derive(Debug, Error)]
 pub enum Esp32PwmError {
     #[error("{0}")]
     EspError(EspError),
-    #[error("only 4 different PWM frequencies allowed, available freq: {0}, {1}, {2}, {3}")]
-    NoTimersAvailable(u32, u32, u32, u32),
+    #[error("all timer are used try a different frequency")]
```

Comment:
are the currently used frequencies not valuable information when receiving this error?

Reply:
It is indeed, however since in RDK (at least for the pi) we do not error out when configuring an infeasible PWM frequency then a user will most likely never see such error. I will add some logic to find the nearest frequency usable